tests/basics.rs

```diff
@@ -5,53 +5,100 @@ use std::fs::File;
 use std::io::Write;
 use std::path::Path;
 
-use goldenfile::Mint;
+use goldenfile::{Mint, DifferType};
 
-fn setup_file(path: &str, contents: &str) {
-    let path = Path::new(path);
+fn open_file(path: &str) -> File{
+let path = Path::new(path);
     fs::create_dir_all(path.parent().unwrap()).unwrap();
-    let mut file = File::create(path).unwrap();
-    write!(file, "{}", contents).unwrap();
+    File::create(path).unwrap()
+}
+
+fn setup_text_file(path: &str, contents: &str) {
+    write!(open_file(path), "{}", contents).unwrap();
+}
+
+fn setup_binary_file(path: &str, buffer: &[u8]) {
+    open_file(path).write(buffer).unwrap();
 }
 
 #[test]
 fn basic_usage() {
-    setup_file("tests/goldenfiles/basic_usage1.txt", "Hello world!");
-    setup_file("tests/goldenfiles/basic_usage2.txt", "foobar");
+    setup_text_file("tests/goldenfiles/basic_usage1.txt", "Hello world!");
+    setup_text_file("tests/goldenfiles/basic_usage2.txt", "foobar");
+    setup_binary_file("tests/goldenfiles/basic_usage3.bin", b"BinaryData\x0f\x0c");
+    setup_binary_file("tests/goldenfiles/basic_usage4.bin", b"BinaryData2\xa0\xa0");
```

Comment:
Rename basic_usage to zero_diff_text and move binary file testing out into a new zero_diff_binary function.